lighthouse-core/config/budget.js

```diff
@@ -52,6 +59,55 @@ class Budget {
     };
   }
 
+  /**
+ * @param {string} path
+ * @return string
+ */
+  static validatePath(path) {
+    if (!path) {
+      throw new Error(`A valid path must be provided`);
+    }
+
+    const hasLeadingSlash = path[0] === '/';
+    const validWildcardQuantity = ((path.match(/\*/g) || []).length <= 1);
+    const validDollarSignQuantity = ((path.match(/\*/g) || []).length <= 1);
+    const validDollarSignPlacement = (path.indexOf('$') === -1) || (path[path.length - 1] === '$');
+
+    const isValid = hasLeadingSlash && validWildcardQuantity
+      && validDollarSignQuantity && validDollarSignPlacement;
+
+    if (!isValid) {
+      throw new Error(`Invalid path ${path}`);
+    }
+    return path;
+  }
+
+  /**
+* @param {string} url
+* @param {string} pattern
+* @return boolean
+*/
+  static urlMatchesPattern(url, pattern) {
```

Comment:
a comment and link to some sort of robots.txt url path matching spec would be nice so we know later what to compare it to for correctness :)